src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -3211,3 +3224,236 @@ def _aks_approuting_update(
+        if not vmss_list:
+            raise ValidationError(f"No VMSS found in the managed resource group {managed_resource_group}!")
+
+        for vmss in vmss_list:
+            vmss_tag = vmss.tags.get("aks-managed-poolName")
+            if vmss_tag and vmss_tag == nodepool_name:
+                vmss_name = vmss.name
+                logger.debug("Select VMSS: %s", vmss_name)
+                break
+        if not vmss_name:
+            raise ValidationError(f"No VMSS pool matched AKS node pool {nodepool_name}!")
+
+        instances = list(compute_client.virtual_machine_scale_set_vms.list(managed_resource_group, vmss_name))
+        if not instances:
+            raise ValidationError(f"No instances found in the VMSS {vmss_name}!")
+
+        instance_id = instances[0].instance_id
+        logger.debug("Select instance id: %s", instance_id)
+    else:
+        index = node_name.find("vmss")
+        if index != -1:
+            vmss_name = node_name[:index + 4]
+            instance_id = int(node_name[index + 4:], base=36)
+            instance_info = compute_client.virtual_machine_scale_set_vms.get(
+                managed_resource_group, vmss_name, instance_id)
+            if not instance_info:
+                raise ValidationError(f"Instance id {instance_id} not found in VMSS {vmss_name}!")
+            _aks_verify_resource(instance_info, CONST_VIRTUAL_MACHINES)
+        else:
+            raise ValidationError(f"Node name {node_name} is invalid!")
+
+    return vmss_name, instance_id
+
+
+def _aks_get_node_name_as(
+        cmd,
+        node_name,
+        managed_resource_group):
+    compute_client = get_compute_client(cmd.cli_ctx)
+
+    if not node_name:
+        print("No node name specified, will randomly select a node from the cluster")
+
+        vm_list = compute_client.virtual_machines.list(managed_resource_group)
+        if not vm_list:
+            raise ValidationError(f"No VM found in the managed resource group {managed_resource_group}!")
+
+        vm_name = ""
+        for vm in vm_list:
+            try:
+                _aks_verify_resource(vm, CONST_VIRTUAL_MACHINES)
+                vm_name = vm.name
+                logger.debug("Select VM: %s", vm_name)
+                break
+            except ValidationError as ex:
+                logger.warning(ex)
+                continue
+
+        if not vm_name:
+            raise ValidationError("No suitable VM found in the managed resource!")
+    else:
+        vm_name = node_name
+        vm_info = compute_client.virtual_machines.get(managed_resource_group, vm_name)
+        if not vm_info:
+            raise ValidationError(f"VM {vm_name} not found in the managed resource group {managed_resource_group}!")
+        _aks_verify_resource(vm_info, CONST_VIRTUAL_MACHINES)
+
+    return vm_name
+
+
+def aks_check_network_outbound(
+        cmd,
+        client,
+        resource_group,
```

Comment:
```suggestion
        resource_group_name,
```
rename other usages in this function accordingly, and you don't need to set `c.argument('resource_group', options_list=['--resource-group', '-g'],...` in `_params.py`

Reply:
oh I see! Thanks! Just updated